rx_signal_processing/dsp.cu

```diff
@@ -8,323 +18,530 @@
-                                                                                     // REPLY I see what your are saying, but I think the better way is through templating.
-                                                                                     // the problem I see here though is that this is the least of our redesign issues we need
-                                                                                     // account for if we want to switch from complex floats.
+/**
+ * @brief      Allocates device memory for the first stage filters and then copies them to the
+ *             device.
+ *
+ * @param[in]  taps        A pointer to the first stage filter taps.
+ * @param[in]  total_taps  The total number of taps for all filters.
+ */
+void DSPCore::allocate_and_copy_first_stage_filters(void *taps, uint32_t total_taps)
 {
-    first_stage_bp_filters_size = total_taps * sizeof(cuComplex);
-    gpuErrchk(cudaMalloc(&first_stage_bp_filters, first_stage_bp_filters_size));
-    gpuErrchk(cudaMemcpyAsync(first_stage_bp_filters, taps,
-                first_stage_bp_filters_size, cudaMemcpyHostToDevice, stream));
+  first_stage_bp_filters_size = total_taps * sizeof(cuComplex);
+  gpuErrchk(cudaMalloc(&first_stage_bp_filters_d, first_stage_bp_filters_size));
+  gpuErrchk(cudaMemcpyAsync(first_stage_bp_filters_d, taps,
+        first_stage_bp_filters_size, cudaMemcpyHostToDevice, stream));
 }
 
-void DSPCore::allocate_and_copy_second_stage_filters(void *taps, uint32_t total_taps)
+/**
+ * @brief      Allocates device memory for the second stage filters and then copies them to the
+ *             device.
+ *
+ * @param[in]  taps        A pointer to the second stage filter taps.
+ * @param[in]  total_taps  The total number of taps for all filters.
+ */
+void DSPCore::allocate_and_copy_second_stage_filter(void *taps, uint32_t total_taps)
 {
-    second_stage_filters_size = total_taps * sizeof(cuComplex);
-    gpuErrchk(cudaMalloc(&second_stage_filters, second_stage_filters_size));
-    gpuErrchk(cudaMemcpyAsync(second_stage_filters, taps,
-               second_stage_filters_size, cudaMemcpyHostToDevice, stream));
+  second_stage_filter_size = total_taps * sizeof(cuComplex);
+  gpuErrchk(cudaMalloc(&second_stage_filter_d, second_stage_filter_size));
+  gpuErrchk(cudaMemcpyAsync(second_stage_filter_d, taps,
+         second_stage_filter_size, cudaMemcpyHostToDevice, stream));
 }
 
-void DSPCore::allocate_and_copy_third_stage_filters(void *taps, uint32_t total_taps)
+/**
+ * @brief      Allocates device memory for the third stage filters and then copies them to the
+ *             device.
+ *
+ * @param[in]  taps        A pointer to the third stage filters.
+ * @param[in]  total_taps  The total number of taps for all filters.
+ */
+void DSPCore::allocate_and_copy_third_stage_filter(void *taps, uint32_t total_taps)
 {
-    third_stage_filters_size = total_taps * sizeof(cuComplex);
```

Comment:
Since you've corrected the 'filters' to 'filter' should then the descriptions for taps and total taps be updated for the second and third stage allocate and copy functions? "A pointer to the third stage filter" "The total number of taps for the third stage filter"